subsys/settings/src/settings_zms_legacy.c

```diff
@@ -0,0 +1,435 @@
+/*
+ * Copyright (c) 2025 BayLibre SAS
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```

Comment:
you cannot just copy an open source file from one repo into another repo and change the License.

The copyright and license of the original file should be kept.

Ref: https://github.com/nrfconnect/sdk-zephyr/blob/06c6605ca152de466718b0cb3f73ae7bba17e6ff/subsys/settings/src/settings_zms.c#L2-L4